scraper.go

```diff
@@ -338,25 +337,28 @@ func newAgentCollector(conf *Config, agentJobHistoryCache AgentJobHistoryCache,
 		}
 		agentCountGauge.Reset()
 		if len(agents) == 0 {
+			log.Println("No agents found")
 			return nil
 		}
-
-		jobStats := [][]string{}
 		for _, a := range agents {
 			resource := getAgentResource(a)
 			agentCountGauge.WithLabelValues(
 				a.BuildState, a.AgentState, a.AgentConfigState, resource,
 			).Add(1)
+		}
 
-			if a.BuildState != "Building" {
-				continue
-			}
+		jobStats := [][]string{}
+		for _, a := range agents {
 			history, err := client.GetJobHistory(a.BuildDetails.PipelineName, a.BuildDetails.StageName, a.BuildDetails.JobName)
 			if err != nil {
 				return err
 			}
+			if a.BuildState != "Building" {
+				continue
+			}
```

Comment:
```suggestion
			}
			if a.BuildState != "Building" {
				continue
			}
			history, err := client.GetJobHistory(a.BuildDetails.PipelineName, a.BuildDetails.StageName, a.BuildDetails.JobName)
			if err != nil {
				return err
```
This is the order of these two blocks in the old code. We don't use the history result if the `BuildState` is not `Building`. The only impact the new code change will give is that if failing to get the history for that agent we will return an error.
Maybe we should instead of returning an error just log and continue with the next agent.